Bottom: 91e2dabf442b824abdafea87f097faaf471d6a94
Top:    73e0207654b9f29c8b464d207369fde84841d119
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-21 20:10:07 +0000

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index e28f584..dfcf4b6 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -227,6 +227,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
